src/components/ResizeDetector.js

```diff
@@ -0,0 +1,204 @@
+import React, {
+  PureComponent, isValidElement, cloneElement,
+} from 'react';
+import { findDOMNode } from 'react-dom';
+import {
+  bool, number, string, shape, func, any,
+} from 'prop-types';
+import ResizeObserver from 'resize-observer-polyfill';
+import rafSchd from 'raf-schd';
+import { getHandle, isFunction, checkIsHaveWindow } from 'lib/utils';
+import Reference from 'components/Reference';
+
+class ResizeDetector extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    const {
+      skipOnMount,
+      refreshMode,
+      refreshRate,
+      refreshOptions,
+    } = props;
+
+    this.state = {
+      width: undefined,
+      height: undefined,
+    };
+
+    this.skipOnMount = skipOnMount;
+    this.raf = null;
+    this.element = null;
+    this.unmounted = false;
+
+    const handle = getHandle(refreshMode);
+    this.resizeHandler = handle
+      ? handle(this.createResizeHandler, refreshRate, refreshOptions)
+      : this.createResizeHandler;
+
+    this.resizeObserver = new ResizeObserver(this.resizeHandler);
+  }
+
+  componentDidMount() {
+    this.toggleObserver(true);
+  }
+
+  componentWillUnmount() {
+    this.toggleObserver(false);
+
+    this.rafClean();
+
+    this.cancelHandler();
+
+    this.unmounted = true;
+  }
+
+  cancelHandler = () => {
+    const shouldCancel = !!(this.resizeHandler && this.resizeHandler.cancel);
+    if (shouldCancel) {
+      // cancel debounced handler
+      this.resizeHandler.cancel();
+      this.resizeHandler = null;
+    }
+  }
+
+  rafClean = () => {
+    const shouldClean = !!(this.raf && this.raf.cancel);
+    if (shouldClean) {
+      this.raf.cancel();
+      this.raf = null;
+    }
+  }
+
+  toggleObserver = (isOn) => {
+    const element = this.getElement();
+    if (!element) return false;
+
+    const type = isOn ? 'observe' : 'unobserve';
+    const handle = this.resizeObserver[type];
+    if (!handle) return false;
+
+    this.resizeObserver[type](element);
+    return true;
```

Comment:
let's remove this `return` and replace
`if (!element) return false;` -> `if (!element) return;`

Reply:
thanks for your suggest